librz/util/mem.c

```diff
@@ -36,6 +36,11 @@ RZ_API void rz_mem_copyloop(ut8 *dest, const ut8 *orig, int dsize, int osize) {
 	}
 }
 
+RZ_API void *rz_mem_copy(void *dest, size_t dmax, const void *src, size_t smax) {
+	rz_return_val_if_fail (dest && src, NULL);
```

Comment:
```suggestion
	rz_return_val_if_fail ((smax == 0 || dmax == 0) || (dest && src), NULL);
```

Reply:
Also, after this  you could just add an `if (smax == 0 || dmax == 0)` to skip the `memcpy` call at all, if you want. By doing this we don't need to have an `if (...)` at every place that calls `rz_mem_cpy`

Reply:
The suggested change will print an unnecessary assertion even though smax/dmax=0 doesn't warrant a warning, added a simple if on entry.

Reply:
yeah :D